build_metal.sh

```diff
@@ -18,8 +18,8 @@ else
 fi
 
 echo "Building tt-metal"
-cmake -B build
-cmake --build build -- -j`nproc`
```

Comment:
If we want this in asap, I think we should have a bash check here to see if `ninja` is available on the host system. If so, then set this: https://cmake.org/cmake/help/latest/envvar/CMAKE_GENERATOR.html#envvar:CMAKE_GENERATOR

If not, default to make so build doesn't just refuse to build.

I think we still need to install `ninja-build` on all the CI and dev machines that are under our purview. + related changes in MIW playbooks.

Reply:
I don't see the benefit of setting `CMAKE_GENERATOR=Ninja` vs. `-G Ninja`. We shouldn't enforce that ninja be used, rather it's preferred over `make`.

Reply:
I'll install `ninja` on our CI machines and use it in our pipelines in a separate PR

Reply:
👍